src/coreclr/vm/object.h

```diff
@@ -555,6 +555,15 @@ class ArrayBase : public Object
     // INT32      lowerBounds[rank];  Valid indexes are lowerBounds[i] <= index[i] < lowerBounds[i] + bounds[i]
 
 public:
+#ifndef DACCESS_COMPILE
+    void SetMethodTableAndNumComponents(MethodTable *pMT, INT32 length)
```

Comment:
I think it is better to keep this as two separate calls. 

There are multiple ways to set the MethodTable (`SetMethodTable` vs. `SetMethodTableForUOHObject`) so combining the two would lead to `SetMethodTableForUOHObjectAndNumComponents` that does not look pretty.

It would be nice to change the two places in gchelpers.cpp to use the setter method instead of direct field access, and delete the `friend`s on `ArrayBase`.

It would be nice to rename `set_EEType` and `InitArrayLength` in nativeaot to use the same names to keep aligning with the shared implementation.